src/EnergyPlus/BaseboardRadiator.cc

```diff
@@ -843,9 +841,14 @@ namespace BaseboardRadiator {
                             SimHWConvective(state, baseboardNum, LoadMet);
 
                             if (LoadMet > DesCoilLoad) { // if the load met is greater than design load, OK to iterate on UA
-                                // Invert the baseboard model: given the design inlet conditions and the design load,
-                                // find the design UA.
-                                General::SolveRoot(state, Acc, MaxIte, SolFla, UA, HWBaseboardUAResidual, UA0, UA1, Par2);
+                                // Invert the baseboard model: given the design inlet conditions and the design load, find the design UA.
+                                auto f = [&state, &baseboardNum, &DesCoilLoad](Real64 UA) {
```

Comment:
With this approach, a lambda is created that captures variables which used to be coerced into the intermediate Par array.  Now they can be accessed directly inside the lambda, no need to cast floats to ints back and forth, just access the variables directly by name.  Much cleaner.  The lambda should capture all the necessary parameters, including `&state`, and then accept just one floating point X argument.